Core/src/org/sleuthkit/autopsy/logicalimager/dsp/AddLogicalImageTask.java

```diff
@@ -363,47 +427,36 @@ private void addInterestingFiles(Path resultsPath, boolean createVHD) throws IOE
                 String parentPath = fields[8];
 
                 if (lineNumber % 100 == 0) {
-                    progressMonitor.setProgressText(Bundle.AddLogicalImageTask_addingInterestingFile(lineNumber, totalFiles));
+                    progressMonitor.setProgressText(Bundle.AddLogicalImageTask_searchingInterestingFile(lineNumber, totalFiles));
                 }
-                String query = makeQuery(createVHD, vhdFilename, fileMetaAddressStr, parentPath, filename);
 
-                // TODO - findAllFilesWhere should SQL-escape the query
+                String query = makeQuery(vhdFilename, fileMetaAddressStr, parentPath, filename);
                 List<AbstractFile> matchedFiles = Case.getCurrentCase().getSleuthkitCase().findAllFilesWhere(query);
+                List<Long> fileIds = new ArrayList<>();
                 for (AbstractFile file : matchedFiles) {
-                    addInterestingFileToArtifacts(file, ruleSetName, ruleName, artifacts);
+                    fileIds.add(file.getId());
+                }
+                String key = String.format("%s\t%s", ruleSetName, ruleName);
+                List<Long> value = new ArrayList<>();
+                if (interestingFileMap.containsKey(key)) {
+                    value = interestingFileMap.get(key);
```

Comment:
This might be simpler as:
if (interestingFileMap.containsKey(key)) {
 interestingFileMap.get(key).addAll(fileIds);
} else {
 interestingFileMap.put(key, fileIds);
}

Reply:
Fixed